File: 2021/2/2.py
with open("2.in") as input_file:
    hor = 0
    depth = 0
    for line in input_file:
        dir, dist = line.strip().split()
        dist = int(dist)
        if dir == 'forward':
            hor += dist
        elif dir == 'down':
            depth += dist
        else:
            depth -= dist
    print(hor*depth)

with open("2.in") as input_file:
    aim = 0
    hor = 0
    depth = 0
    for line in input_file:
        dir, dist = line.strip().split()
        dist = int(dist)
        if dir == 'down':
            aim += dist
        elif dir == 'up':
            aim -= dist
        else:
            hor += dist
            depth += aim * dist
    print(hor*depth)

File: 2021/2/2.rb
lines = File.readlines("2.in")

hor = 0
depth = 0
lines.map(&:chomp).each do |line|
  dir, dist = line.split
  dist = dist.to_i
  hor += dist   if dir == "forward"
  depth -= dist if dir == "up"
  depth += dist if dir == "down"
end
puts hor*depth


# Task 2
aim = 0
hor = 0
depth = 0
lines.map(&:chomp).each do |line|
  dir, dist = line.split
  dist = dist.to_i

  aim += dist if dir == "down"
  aim -= dist if dir == "up"

  if dir == "forward"
    hor += dist
    depth += aim * dist
  end
end
puts hor*depth
